kernel/Makefile

```diff
@@ -2,98 +2,113 @@ include ../mk/build.mk
 
 EXTRA_CFLAGS=-I.
 
+KERNEL_ELF=cosmos.elf
+KERNEL_BIN=cosmos.bin
+
 all: kernel
 
 kernel: cosmos
 
-cosmos: cosmos.o asm-dir ata-dir console-dir interrupts-dir keyboard-dir mm-dir panic-dir pci-dir string-dir timing-dir video-dir
-	$(LD) $(LDFLAGS) -o cosmos \
+cosmos: $(KERNEL_ELF)
+	$(OBJCOPY) $(OBJCOPYFLAGS) $(KERNEL_ELF) $(KERNEL_BIN)
```

Comment:
Is there an advantage to doing it like this?

Reply:
yup.  There are debuggers that can read ELF binaries and debug into them.